From b0970498f9de9566032e7f6d9386960cf96a3c04 Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Thu, 16 Nov 2023 17:21:20 +0100
Subject: [PATCH 0622/1141] arm64: dts: st: add UART console on stm32mp257f-dk
 boards

Add USART2 to support console on stm32mp257f-dk boards.

Signed-off-by: Olivier BIDEAU <olivier.bideau@st.com>
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Change-Id: I3fe263b9dfd70401f3095ba407180904c480c853
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/340495
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index dc3a4fe587d5..f83308f8f9a5 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -17,6 +17,7 @@ / {
 	compatible = "st,stm32mp257f-dk", "st,stm32mp257";
 
 	aliases {
+		serial0 = &usart2;
 	};
 
 	chosen {
@@ -53,3 +54,13 @@ &sdmmc1 {
 	vqmmc-supply = <&scmi_vddio1>;
 	status = "okay";
 };
+
+&usart2 {
+	pinctrl-names = "default", "idle", "sleep";
+	pinctrl-0 = <&usart2_pins_a>;
+	pinctrl-1 = <&usart2_idle_pins_a>;
+	pinctrl-2 = <&usart2_sleep_pins_a>;
+	/delete-property/dmas;
+	/delete-property/dma-names;
+	status = "okay";
+};
-- 
2.39.2

